File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cs
public class Solution {
    public int NumSubarrayProductLessThanK(int[] nums, int k) {
        if(k <= 1) return 0;
        
        int cnt = 0;
        int currProd = 1;
        int left = 0;
        
        for(int right=0; right < nums.Length; right++)
        {
            currProd *= nums[right];
            while(currProd >= k)
            {
                currProd /= nums[left];
                left++;
            }
            
            cnt += right-left+1;
            
        }
        
        return cnt;
        
    }
}

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.ts
function numSubarrayProductLessThanK(nums: number[], k: number): number {
    if(k <= 1) return 0;
    
    let cnt: number = 0;
    let currProd: number = 1;
    let left: number = 0;
    
    for(let right=0; right < nums.length; right++){
        currProd *= nums[right];
        
        while(currProd >= k){
            currProd /= nums[left];
            left++;
        }
        cnt += right-left+1;
    }
    return cnt;
};